app/Http/Controllers/GetSubmissionsController.php

```diff
@@ -17,7 +17,14 @@ class GetSubmissionsController
      */
     public function __invoke(Request $request): JsonResponse
     {
-        $submissions = $request->user()->submissions()->paginate();
+        $status = $request->input('status');
+        $submissions = $request
+            ->user()
+            ->submissions()
+            ->when($status, function ($query, $status) {
+                return $query->where('status', $status);
+            })
+            ->paginate();
 
         return response()->json($submissions);
```

Comment:
```suggestion
        return $request
            ->user()
            ->submissions()
            ->when($status, function ($query, $status) {
                return $query->where('status', $status);
            })
            ->paginate();
```

Podes retornar directamente el resultado de la paginacion dado que Laravel se encarga de convertirlo en una respuesta JSON. Los modelos tambien pueden ser retornados asi como tal, y Laravel se va a encargar de convertirlo a JSON. Aprovechemos que el Framework se encarga de cosas como esa por nosotros.